crates/payload/basic/src/builder_stack.rs

```diff
@@ -0,0 +1,311 @@
+use crate::{
+    BuildArguments, BuildOutcome, PayloadBuilder, PayloadBuilderError,
+    PayloadConfig, PayloadBuilderAttributes
+};
+use futures_util::future::Either as EitherFuture;
+
```

Comment:
While it might be code duplication, we're not really implementing `Future` for an either type here, so it seems better to just hand roll an `Either` type for the crate and implement `Error` / `PayloadBuilderAttributes` / `BuiltPayload` on it. It would help naming as well